src/controllers/openai.controller.ts

```diff
@@ -0,0 +1,12 @@
+import { Controller, Post, Body } from "@nestjs/common";
+import { OpenAIService } from "../services/openai.service";
+
+@Controller("openai")
+export class OpenAIController {
+    constructor(private readonly openaiService: OpenAIService) {}
+
+    @Post("troubleshooting-steps")
+    async getTroubleshootingSteps(@Body("supportNote") prompt: string): Promise<string> {
```

Comment:
Why there's no use of DTO here? Also this isn't a prompt its a chat conversation summary.